### PR TITLE
E2E test for Create note and App routing

### PR DESCRIPTION
**Pull request by AliTaee**

### CODE REVIEW

File: cypress/integration/createNote.test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/* eslint-env jest */
```
Comment (sohrabtaee) [inline cypress/integration/createNote.test.js]: You can create an `.eslintrc` file and move this rule and any other rules there instead of repeating it as a comment in multiple files.

File: cypress/integration/createNote.test.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+describe('Create Note test', () => {
+  it('Go to home page', () => {
```
Comment (sohrabtaee) [inline cypress/integration/createNote.test.js]: Usually, when you use `it` function, it comes with a BDD style statement.E.g. `it('should go to home page', () => {})`.

File: cypress/integration/createNote.test.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    it('App should have a note text: Learning e2e', () => {
+      cy.contains(noteTitle);
+    });
+
+    it('Note list should have one node', () => {
+      cy.get('[data-test=note-item]').should('have.length', 1);
+    });
+  });
+
+  describe('Creating a note with just a title.', () => {
```
Comment (sohrabtaee) [inline cypress/integration/createNote.test.js]: Another example of BDD style testing `it('should create a note with just a title', () => {})`

Comment (sohrabtaee) [reply cypress/integration/createNote.test.js]: Also, when you mention "just a title", it implied that there should not be a body text, so it would be better to check if the body text is empty, too.

File: src/components/note/viewChildComponents/NoteList.jsx [hunk 4]
```diff
@@ -71,6 +71,7 @@ const NoteList = props => {
                 id={item.id}
                 primary={item.title}
                 secondary={`${item.date}`}
+                secondaryTypographyProps={{ 'data-test': 'note-date' }}
```
Comment (sohrabtaee) [inline src/components/note/viewChildComponents/NoteList.jsx]: `secondaryTypographyProps` seems irrelevant to `data-test` attribute. It's not a typography prop!

File: cypress/support/commands.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+// ***********************************************
```
Comment (sohrabtaee) [inline cypress/support/commands.js]: What's the point of having a file with just some comments?

File: cypress/integration/listNote.test.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+      cy.get('[data-test=note-desc]')
+        .type(`${descTitle}-${index}`)
+        .should('have.value', `${descTitle}-${index}`);
+    });
+
+    it('Click on submit', () => {
+      cy.get('[data-test=submit-note]').click();
+    });
+
+    it(`App should have a note title: ${noteTitle}-${index}`, () => {
```
Comment (sohrabtaee) [inline cypress/integration/listNote.test.js]: Test titles should be human-readable (not developer-readable)

File: cypress/integration/listNote.test.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    cy.contains('No notes yet.');
+  });
+});
+
+describe('Create a multi note fot test', () => {
+  const noteTitle = 'test title';
+  const descTitle = 'test desc note';
+
+  // Create 3 sample note test
+  for (let index = 1; index <= 5; index++) {
```
Comment (sohrabtaee) [inline cypress/integration/listNote.test.js]: The comment says 3 notes but the loop repeats 4 times.

File: cypress/integration/editNote.test.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    it('Edited note must have a new title in list.', () => {
+      cy.get('[data-test=note-item]')
+        .first()
+        .contains(newTitle);
+    });
+  });
+
+  describe('Test change new title and description', () => {
+    it('Click on edit note', () => {
+      cy.get('[aria-label=edit]').click();
```
Comment (sohrabtaee) [inline cypress/integration/editNote.test.js]: Always try to use `data-test` attributes for testing because maybe in the future you decide to change the `aria-label` for some accessibility reasons, and it should not break the tests.
To sum it up, changing the implementation details should not break the tests.

File: cypress/integration/createNote.test.js (hunk 1 above)
Comment (AliTaee) [reply cypress/integration/createNote.test.js]: Done.

File: cypress/integration/createNote.test.js (hunk 2 above)
Comment (AliTaee) [reply cypress/integration/createNote.test.js]: Done.

File: cypress/integration/listNote.test.js (hunk 7 above)
Comment (AliTaee) [reply cypress/integration/listNote.test.js]: Done. I was forgetting to correct this comment.

File: cypress/integration/listNote.test.js (hunk 6 above)
Comment (AliTaee) [reply cypress/integration/listNote.test.js]: I have thought it's a good idea to user see exactly the title it test creates.

Comment (AliTaee) [reply cypress/integration/listNote.test.js]: Done.

File: cypress/support/commands.js (hunk 5 above)
Comment (AliTaee) [reply cypress/support/commands.js]: I was using this file in previous commits, then I refactor and deleted code inside. I forgetting to delete this file. Thanks. Done.

File: cypress/integration/editNote.test.js (hunk 8 above)
Comment (AliTaee) [reply cypress/integration/editNote.test.js]: Good point. Done.

File: src/components/note/viewChildComponents/NoteList.jsx (hunk 4 above)
Comment (AliTaee) [reply src/components/note/viewChildComponents/NoteList.jsx]: Done.

File: cypress/integration/createNote.test.js (hunk 3 above)
Comment (AliTaee) [reply cypress/integration/createNote.test.js]: Done.
